doc/src/manual/running-external-programs.md

```diff
@@ -308,6 +308,27 @@ IO redirection can be accomplished by passing keyword arguments `stdin`, `stdout
 pipeline(`do_work`, stdout=pipeline(`sort`, "out.txt"), stderr="errs.txt")
 ```
 
+For example, you can pass a `IOBuffer`, and later read from it, after resetting the stream back to its beginning:
+
+```julia
+julia> io = IOBuffer();
+
+julia> run(pipeline(`echo world`; stdout=io));
+
+julia> seekstart(io);
```

Comment:
The seekstart is also only needed because you are reading into a file-like object. If you use a stream-like object, seeking it is not necessary.
```suggestion
julia> io = PipeBuffer();

julia> run(`echo world`, devnull, io, stderr);
```

Is it worth pointing out here that it is actively very dangerous to read from this type of `io` object before the process has cleanly exited (e.g. `wait=true`), or in any Julia version before v1.11?

Reply:
Why is that very dangerous before v1.11?

Reply:
It is a data race and can therefore lead to arbitrary memory corruption

Reply:
What is the data race on? :thinking: If it's fixed in 1.11, this should definitely be mentioned somewhere.